WordPressKitTests/WordPressComRestApiTests+Locale.swift

```diff
@@ -1,272 +1,98 @@
 import Foundation
 import XCTest
+import OHHTTPStubs
 
 import WordPressShared
 @testable import WordPressKit
 
 extension WordPressComRestApiTests {
 
-    func testThatAppendingLocaleWorks() {
-        // Given
-        let path = "/path/path"
-        let preferredLanguageIdentifier = WordPressComLanguageDatabase().deviceLanguage.slug
+    func testThatAppendingLocaleWorks() async throws {
+        var request: URLRequest?
+        stub(condition: { _ in true }, response: {
+            request = $0
+            return HTTPStubsResponse(error: URLError(.networkConnectionLost))
+        })
 
-        // When
         let api = WordPressComRestApi()
-        let localeAppendedPath = api.buildRequestURLFor(path: path)
-
-        // Then
-        XCTAssertNotNil(localeAppendedPath)
-        let actualURL = URL(string: localeAppendedPath!, relativeTo: api.baseURL)
-        XCTAssertNotNil(actualURL)
-
-        let actualURLComponents = URLComponents(url: actualURL!, resolvingAgainstBaseURL: false)
-        XCTAssertNotNil(actualURLComponents)
-
-        let expectedPath = path
-        let actualPath = actualURLComponents!.path
-        XCTAssertEqual(expectedPath, actualPath)
-
-        let actualQueryItems = actualURLComponents!.queryItems
-        XCTAssertNotNil(actualQueryItems)
-
-        let expectedQueryItemCount = 1
-        let actualQueryItemCount = actualQueryItems!.count
-        XCTAssertEqual(expectedQueryItemCount, actualQueryItemCount)
-
-        let actualQueryItem = actualQueryItems!.first
-        XCTAssertNotNil(actualQueryItem!)
-
-        let actualQueryItemKey = actualQueryItem!.name
-        let expectedQueryItemKey = WordPressComRestApi.LocaleKeyDefault
-        XCTAssertEqual(expectedQueryItemKey, actualQueryItemKey)
+        let _ = await api.perform(.get, URLString: "/path/path")
 
-        let actualQueryItemValue = actualQueryItem!.value
-        XCTAssertNotNil(actualQueryItemValue)
-
-        let expectedQueryItemValue = preferredLanguageIdentifier
-        XCTAssertEqual(expectedQueryItemValue, actualQueryItemValue!)
+        let preferredLanguageIdentifier = WordPressComLanguageDatabase().deviceLanguage.slug
+        XCTAssertEqual(request?.url?.query, "locale=\(preferredLanguageIdentifier)")
     }
 
-    func testThatAppendingLocaleWorksWithExistingParams() {
-        // Given
+    func testThatAppendingLocaleWorksWithExistingParams() async throws {
```

Comment:
What do you think of

```suggestion
    func testAddsLocaleToURLQueryByDefaultAndMaintainsInputParameters() async throws {
```